src/tirith/core/core.py

```diff
@@ -81,8 +81,18 @@ def generate_evaluator_result(evaluator_obj, input_data, provider_module):
                 has_evaluation_passed = None
                 continue
 
+            # Run evaluation on the provider's input value
             evaluation_result = evaluator_instance.evaluate(evaluator_input["value"], evaluator_data)
-            evaluation_result["meta"] = evaluator_input.get("meta")
+
+            # Copy metadata and addresses if provided by the provider
+            if "meta" in evaluator_input:
+                evaluation_result["meta"] = evaluator_input["meta"]
+
+            if "addresses" in evaluator_input:
+                # Add addresses directly
+                # TODO: We need to make a model class for the `evaluator_input` and move this validation there
```

Comment:
Please add the type assertion mentioned in https://github.com/StackGuardian/tirith/pull/262#discussion_r2142707826 here
```py
                if not isinstance(addresses, list):
                    raise Exception("`addresses` should be a list")
```